### PR TITLE
refact: don't authZ User and instead filter the returned roles

### PR DESCRIPTION
**Pull request by moogacs**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/handlers/rest/handlers_authz.go [hunk 1]
```diff
@@ -319,28 +318,36 @@ func (h *authZHandlers) getRolesForUser(params authz.GetRolesForUserParams, prin
...
 		perms, err := conv.PoliciesToPermission(policies...)
 		if err != nil {
 			return authz.NewGetRolesInternalServerError().WithPayload(errPayloadFromSingleErr(err))
 		}
+
+		// verify the user has access
+		if err := h.authorizer.Authorize(principal, authorization.READ, authorization.Roles(roleName)...); err != nil {
+			authErr = err
+			continue
+		}
```
Comment (dirkkul) [inline adapters/handlers/rest/handlers_authz.go:339]: wouldn't it be easier to just return a forbidded error here? 🤔

Comment (moogacs) [reply adapters/handlers/rest/handlers_authz.go:339]: nope, because there could be many roles and the users has access to 1 of them only

Comment (dirkkul) [reply adapters/handlers/rest/handlers_authz.go:339]: so we return everything that the caller has access to and ignore all authErrors unless there is not a single role the caller can access?

Comment (moogacs) [reply adapters/handlers/rest/handlers_authz.go:339]: exactly 

let's the user has many roles 
- RoleA
- RoleB 
 but the user requesting this request has access to only RoleA to read
 that's why we decide on length https://github.com/weaviate/weaviate/pull/6427/files#diff-94644b2fbe6ecbfccdc283fb7d87a3386c6730ddc08423e63554ed48d7ad62a6R347

**Review by dirkkul — APPROVED** [review]

**Pull request merged**